test/test_variable_size_binary_layout.cpp

```diff
@@ -0,0 +1,131 @@
+// Copyright 2024 Man Group Operations Limited
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "doctest/doctest.h"
+
+#include <numeric>
+#include <string_view>
+
+#include "sparrow/array_data.hpp"
+#include "sparrow/variable_size_binary_layout.hpp"
+
+#include <iostream>
+
+namespace sparrow
+{
+    struct vs_binary_fixture
+    {
+        vs_binary_fixture()
+        {
+            m_data.bitmap.resize(nb_words);
+            m_data.buffers.resize(2);
+            m_data.buffers[0].resize(sizeof(std::int64_t) * (nb_words + 1));
+            m_data.buffers[1].resize(std::accumulate(
+                words, words + nb_words, 0u, [](std::size_t res, const auto& s) { return res + s.size(); }
+            ));
+            m_data.buffers[0].data<std::int64_t>()[0] = 0u;
+            auto iter = m_data.buffers[1].begin();
+            for (size_t i = 0; i < nb_words; ++i)
+            {
+                offset()[i+1] = offset()[i] + words[i].size();
+                std::copy(words[i].cbegin(), words[i].cend(), iter);
+                iter += words[i].size();
+                m_data.bitmap.set(i, true);
+            }
+
+            m_data.bitmap.set(2, false);
+            m_data.length = 4;
+            m_data.offset = 1;
+        }
+
+        static constexpr size_t nb_words = 4u;
+        static constexpr std::string_view words[nb_words] = 
+        {
+            "you",
+            "are",
+            "not",
+            "prepared"
+        };
+
+        array_data m_data;
+        // TODO: replace R = std::string_view with specific reference proxy
+        using layout_type = variable_size_binary_layout<std::string, std::string_view, std::string_view>;
+    
+    private:
+
+        std::int64_t* offset()
+        {
+            return m_data.buffers[0].data<std::int64_t>();
+        }
+    };
+
+    TEST_SUITE("variable_size_binary_layout")
+    {
+        TEST_CASE_FIXTURE(vs_binary_fixture, "types")
```

Comment:
No need for a runtime test case when it's a bunch of compile-time checks.

I wonder if there is compile-time checks in doctest, I know there are in catch2 (I prefer doctest usually)

Reply:
Note that if any static assert needs to use that class, then the class must be defined before that use, so probably it's best to put them next but after that class.

Reply:
The static asserts are on the layout_type defined in the fixture class, so I guess it's fine to keep the static assert in the fixture class?